### ISSUE TITLE
Show QR code

### ISSUE THREAD

**Issue by JohnBillington** [feross/instant.io#94]
Show a QR code for the different links so that mobile devices can become peers without users having to type in the URL.

**Comment by joshterrill** [comment]
I can give this a go if it's wanted.

**Comment by JohnBillington** [comment]
I only use instant.io occasionally, so don't feel obliged to do this for my sake. I do think that this feature would be very handy though.

**Comment by feross** [comment]
@joshterrill Yeah, that would be cool to try. But are there any good QR *reading* libraries in pure javascript? I've seen tons of QR code generators, but only a few that can take grainy images from a webcam and parse the QR codes out of them.

**Comment by joshterrill** [comment]
I'll look around and post what I find before I really start digging into it.

**Comment by xxxorg** [comment]
@feross I found just https://github.com/LazarSoft/jsqrcode, do you know any other qr reading in js?

**Comment by xxxorg** [comment]
Would you accept PR just for the qr code generation (not the reading)?

**Comment by joshterrill** [comment]
@xxxorg I don't think we need it to read... I think it's just QR code generation so that a mobile app who has a reader can scan it. Am I right?

**Comment by feross** [comment]
I would accept a PR just for generation for now. We can figure out reading
later.

**Comment by nodiscc** [comment]
I can recommend http://neocotic.com/qr.js/ for QR generation which I have been using for years in a web application.

**Comment by joshterrill** [comment]
I just wrote this today but forgot to commit it. I'll do it later today.

**Comment by rugk** [comment]
As for qrcode generation there is also https://larsjung.de/kjua/. It works without jQuery and is very powerful.

**Comment by aswinmohanme** [comment]
@feross can you look into this ?

**Comment by joshterrill** [comment]
I apologize for letting this just stay open like this. I've been pretty swamped at work. If the suggestion @rugk has looks good to everyone I can implement it over the 4th of July break this week. @feross since that is a different library and the other pull request is pretty old, should we close the other one out and start a new one? Or should I pull the existing one down and edit it?

**Comment by joshterrill** [comment]
@feross I just finished writing the code... not sure how you want me to go about pushing it though. Should I pull down my other PR and modify that? Or use a new one since I'm using a different library than before?

**Comment by joshterrill** [comment]
There were enough differences in the existing code base and what was in that old PR that I figured I could just create a new PR. If this isn't what you want to do, just let me know. I've created https://github.com/webtorrent/instant.io/pull/176 using the library @rugk suggested.

**Comment by ThelloD** [comment]
Any update regarding the implementation status of QR codes? Feature request was created 5 years ago and pull request 1.5 years ago, so it's a bit sad that this wasn't added, yet.

**Comment by pojntfx** [comment]
I would also really appreciate this feature. https://webwormhole.io/ is fantastic in terms of this; if one wants to, for example, transfer a file from their desktop to their phone, this would be greatly appreciated.
